src/lib/datamodel/Attribute.h

```diff
@@ -0,0 +1,117 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file contains definitions for working with CHIP values.
+ *
+ */
+
+#ifndef CHIP_ATTRIBUTE_H_
+#define CHIP_ATTRIBUTE_H_
+
+#include <datamodel/Deque.h>
+#include <datamodel/Value.h>
+
+#include <core/CHIPError.h>
+
+namespace chip {
+namespace DataModel {
+
+/**
+ * @brief
+ *   This class implements a single attribute.
+ */
+class Attribute
+{
+private:
+    Deque<Attribute> mDeque;
+    friend class Cluster;
+
+public:
+    uint16_t mAttrId;
+    Value mValue;
+    Value mMin;
```

Comment:
Attributes that don't have a concept of min/max, or don't even specify such a constraint in schema will pay the cost of storage of these fields.

Reply:
Fixed, these only will show up for Attributes that are range-bound